lib/screens/v2/widget_instance/subway_status.ex

```diff
@@ -0,0 +1,625 @@
+    length(alert_routes(alert)) > 1
+  end
+
+  def serialize_green_line(grouped_alerts) do
+    green_line_alerts =
+      ["Green-B", "Green-C", "Green-D", "Green-E"]
+      |> Enum.flat_map(fn route -> Map.get(grouped_alerts, route, []) end)
+      |> Enum.uniq()
+
+    multi_route_alerts = Enum.filter(green_line_alerts, &is_multi_route?/1)
+    single_route_alerts = Enum.reject(green_line_alerts, &is_multi_route?/1)
+    trunk_alerts = Enum.filter(multi_route_alerts, &alert_affects_gl_trunk?/1)
+    alert_count = length(green_line_alerts)
+
+    statuses =
+      single_route_alerts
+      |> Enum.group_by(fn alert -> Enum.at(alert_routes(alert), 0) end)
+      |> Enum.map(fn {route, alerts} -> {route, get_green_line_branch_statuses(alerts)} end)
+      |> group_green_line_branch_statuses()
+
+    case trunk_alerts do
+      [] ->
+        # If there are no alerts for the GL trunk, serialize any alerts on the branches
+        serialize_green_line_branch_alerts(statuses, single_route_alerts, alert_count)
+
+      [trunk_alert] ->
+        # If there is a single alert on the GL trunk, combine it with any alerts on the branches
+        serialize_green_line_trunk_alert(trunk_alert, statuses, alert_count)
+
+      _ ->
+        # If there are multiple alerts on the GL trunk, log it and serialize the count
+        _ =
+          Logger.info(
+            "[subway_status_multiple_alerts] route=#{"Green-Trunk"} count=#{alert_count}"
+          )
+
+        %{
+          type: :trunk,
+          route: serialize_route_pill("Green"),
+          status: "#{alert_count} alerts",
+          location: ["mbta.com/alerts/subway", "mbta.com/alerts/subway"]
+        }
+    end
+  end
+
+  defp serialize_green_line_trunk_alert(trunk_alert, statuses, alert_count) do
+    case statuses do
+      [] ->
+        # One GL trunk alert and no branch alerts
+        Map.merge(
+          %{type: :trunk, route: serialize_route_pill("Green")},
+          serialize_alert(trunk_alert, "Green")
+        )
+
+      [status] ->
+        # One GL trunk alert and one GL branch alert
+        trunk_status = get_green_line_branch_status(trunk_alert)
+        %{type: :branch, statuses: [[nil, trunk_status], status]}
+
+      _ ->
+        # One GL trunk alert and 2+ GL branch alerts
+        _ =
+          Logger.info(
+            "[subway_status_multiple_alerts] route=#{"Green-Trunk"} count=#{alert_count}"
+          )
+
+        %{
+          type: :trunk,
+          route: serialize_route_pill("Green"),
+          status: "#{alert_count} alerts",
+          location: ["mbta.com/alerts/subway", "mbta.com/alerts/subway"]
+        }
+    end
+  end
+
+  defp serialize_green_line_branch_alerts(statuses, single_route_alerts, alert_count) do
+    case {single_route_alerts, statuses} do
+      # One GL branch alert
+      {[alert], _} ->
+        [route] = alert_routes(alert)
+
+        Map.merge(
+          %{type: :trunk, route: serialize_route_pill("Green"), branch: route},
```

Comment:
Should type be `:branch`? Or if `:trunk` and `:branch` are more about describing whether there's one or multiple alerts to show, maybe rename them to `:single` + `:multiple`.